atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/KvsMigrationCommand.java

```diff
@@ -173,19 +174,27 @@ public AtlasDbServices connectToServices() throws IOException {
     private KeyValueServiceMigrator getMigrator(AtlasDbServices fromServices, AtlasDbServices toServices)
             throws IOException {
         //TODO: this timestamp will have to be stored for online migration
-        Supplier<Long> migrationTimestampSupplier = Suppliers.ofInstance(toServices
-                .getTimestampService()
-                .getFreshTimestamp());
-        toServices.getTransactionService().putUnlessExists(migrationTimestampSupplier.get(), toServices
-                .getTimestampService()
-                .getFreshTimestamp());
+        long migrationStartTimestamp = fromServices.getTimestampService().getFreshTimestamp();
+        long migrationCommitTimestamp = fromServices.getTimestampService().getFreshTimestamp();
+
+        TimestampService toTimestampService = toServices.getTimestampService();
+        if (!(toTimestampService instanceof PersistentTimestampService)) {
```

Comment:
`ManagedTimestampService` is specific to TimeLock though. Since we don't use fresh timestamps, `TimestampManagementService` should suffice (this is how we do it in automigrations as well). I've changed to that.